internal/plans/deferring/deferred.go

```diff
@@ -66,6 +67,19 @@ type Deferred struct {
 	// all of those options to decide if each instance is relevant.
 	resourceInstancesDeferred addrs.Map[addrs.ConfigResource, addrs.Map[addrs.AbsResourceInstance, *plans.DeferredResourceInstanceChange]]
 
+	// ephemeralResourceInstancesDeferred tracks the ephemeral resource instances
```

Comment:
Don't need to refactor now, but would it make sense to just track all these in `resourceInstancesDeferred`? The address are all the same type and are differentiated by mode, but I don't know if it helps reduce code otherwise or not.

Reply:
I discussed this with @liamcervante and we think that since we serialize the other deferred changes into the plan, differentiating them by mode alone would put us at more risk of accidentally not accounting for ephemeral resources. The extra effort we need to put in now does not feel major to warrant that risk.